src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java

```diff
@@ -178,10 +177,12 @@ private static String sanitizeParam(String param) {
                             }
                         }
                     })
-                    .doOnError(throwable -> {
+                    .onErrorResume(throwable -> {
                         if (!(throwable instanceof CancellationException)) {
                             LOGGER.error(throwable.toString(), throwable);
                         }
+                        stoppedPublisherService.publish(resultContext.getResultUuid(), resultContext.getRunContext().getReceiver());
+                        return resultRepository.delete(resultContext.getResultUuid()).then(Mono.empty());
```

Comment:
.then(Mono.empty())

isn't it the same as just .then() ?

Reply:
`then` return a Mono<Void> which only replays complete and error signals : mono which never contains data
`then(Mono.empty())` return a Mono<T> (mono with data)  which contains no data
it is to execute the delete mono